src/drivers/next-json-cookie/index.ts

```diff
@@ -0,0 +1,67 @@
+import nextCookies from 'cookies-next';
+import AES from 'crypto-js/aes';
+import Utf8 from 'crypto-js/enc-utf8';
+import CacheDriver from '../driver';
+import { CookieValueTypes, OptionsType } from 'cookies-next/lib/types';
+import DEFAULT_COOKIE_OPTIONS from './default-cookie-options';
+import { decryptCookie, encryptCookie } from './encrypt-cookie';
+import JSONValue from '../types/json';
+
+class NextCookieDriver<TmpCookiesObj> extends CacheDriver<typeof nextCookies> {
+
+  #crypto?: typeof AES;
+  #encoder?: typeof Utf8;
+
+  constructor(protected store: typeof nextCookies, crypto?: typeof AES, encoder?: typeof Utf8) {
+    super();
+    this.#crypto = crypto;
+    this.#encoder = encoder;
+  }
+
+  public flush(options?: OptionsType): void {
+    const cookieNamesToFlush = Object.keys(this.store.getCookies(options) ?? {}) ??[];
+    cookieNamesToFlush.forEach((cookieNameToFlush) => this.store.deleteCookie(cookieNameToFlush));
+  }
+
+  public get<T = JSONValue>(key: string, fallback: T | null = null, options?: OptionsType): T | null {
+    if (this.has(key)) {
+      try {
+        const cookieValue = this.store.getCookie(key, options) as string;
+        return decryptCookie(cookieValue, this.#crypto, this.#encoder) as unknown as T;
+      } catch (error) {
+        return fallback;
+      }
+    }
+
+    return fallback;
+  }
+
+  public getAll<T = JSONValue>() {
+    const cookieValues = this.store.getCookies();
+    const allCookies: Record<string, T> = {};
+    Object.keys(cookieValues).forEach((key) => {
+      allCookies[key] = decryptCookie<T>(cookieValues[key] as string);
+    });
+    return allCookies;
```

Comment:
Would work better as a reducer.
```suggestion
    return Object.entries(this.store.getCookies()).reduce((cookies, [name, value]) => {
      return { ...cookies, [name]: decryptCookie(value) };
    }, {});
```